dsf-bpe/dsf-bpe-process-base/src/main/java/org/highmed/dsf/bpe/plugin/AbstractProcessEnginePlugin.java

```diff
@@ -1,19 +1,38 @@
 package org.highmed.dsf.bpe.plugin;
 
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
 import org.camunda.bpm.engine.ProcessEngine;
 import org.camunda.bpm.engine.RepositoryService;
+import org.camunda.bpm.engine.delegate.JavaDelegate;
 import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
 import org.camunda.bpm.engine.impl.cfg.ProcessEnginePlugin;
 import org.camunda.bpm.engine.repository.Deployment;
 import org.camunda.bpm.model.bpmn.Bpmn;
+import org.camunda.bpm.model.bpmn.BpmnModelException;
 import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.Process;
+import org.camunda.bpm.model.bpmn.instance.ServiceTask;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.camunda.bpm.model.xml.type.ModelElementType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.context.ApplicationContext;
 
 public abstract class AbstractProcessEnginePlugin implements ProcessEnginePlugin
 {
 	private static final Logger logger = LoggerFactory.getLogger(AbstractProcessEnginePlugin.class);
 
+	private final ApplicationContext context;
+
+	public AbstractProcessEnginePlugin(ApplicationContext context)
```

Comment:
I'm not sure if we want to hand an ApplicationContext object down to every ProcessEnginePlugin -> information hiding. It might be better to implement a method that, returns a list/stream of all the configured service delegate class names (or a Map<String: ProcessName, List<String: ClassName>>) and implement the bean availability check part within a validator service in the dsf-bpe-server module that has a list of AbstractProcessEnginePlugins as a dependency.

Reply:
implemented as contextlistener service based in contextrefreshed event